src/runner.js

```diff
@@ -180,12 +183,16 @@ exports.run = function(config) {
       timer = setInterval(function() { console.log('.'); }, 120*1000);
       return api.waitForBuild(config.apiKey, config.project, config.branch, config.build).timeout(MAX_MS, 'Timeout waiting for Build');
     })
+    // disconnect tunnel and relay the response
     .then(function(response) {
-      clearInterval(timer);
-      if (config.tunnel) {
-        console.log('Disconnecting tunnel');
-        Tunnel.disconnect();
+      if (config.tunnel || (config.sauce && config.sauce.launchSauceConnect)) {
+        return Tunnel.disconnect(response);
```

Comment:
@xwa130saucer this `response` doesn't have anything to do with the tunnel, so it seems out of place. Can we not pass it in, and change it to something like the following instead?

```
return Tunnel.disconnect()
  .then(function() { return response; });
```

Reply:
Good idea 👍 
changed